agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -572,23 +572,16 @@ bool backhaul_manager::finalize_slaves_connect_state(bool fConnected,
 
         auto db = AgentDB::get();
 
-        std::string strIface;
         network_utils::iface_info iface_info;
         bool backhaul_manager_exist = false;
 
         notification->params().is_prplmesh_controller = is_prplmesh_controller;
         notification->params().controller_bridge_mac = tlvf::mac_from_string(controller_bridge_mac);
 
         if (!db->device_conf.local_gw) {
-
-            if (db->backhaul.connection_type == AgentDB::sBackhaul::eConnectionType::Wired) {
-                strIface = db->ethernet.iface_name;
-            } else {
-                strIface = m_sConfig.wireless_iface;
-            }
             // Read the IP addresses of the bridge interface
-            if (network_utils::get_iface_info(iface_info, strIface) != 0) {
-                LOG(ERROR) << "Failed reading addresses for: " << strIface;
+            if (network_utils::get_iface_info(iface_info, db->backhaul.selected_iface_name) != 0) {
+                LOG(ERROR) << "Failed reading addresses for: " << db->backhaul.selected_iface_name;
```

Comment:
```suggestion
            auto iface_name = db->backhaul.selected_iface_name;
            if (network_utils::get_iface_info(iface_info, iface_name) != 0) {
                LOG(ERROR) << "Failed reading addresses for: " << iface_name;
```

you are using the `db->backhaul.selected_iface_name` several times here - use a local variable